tests/ion_tests/mod.rs

```diff
@@ -47,19 +50,19 @@ pub fn serialize(format: Format, elements: &Sequence) -> IonResult<Vec<u8>> {
     let mut buffer = Vec::with_capacity(2048);
     match format {
         Format::Text(kind) => {
-            let mut writer = match kind {
-                TextKind::Compact => TextWriterBuilder::default().build(&mut buffer),
-                TextKind::Lines => TextWriterBuilder::lines().build(&mut buffer),
-                TextKind::Pretty => TextWriterBuilder::pretty().build(&mut buffer),
-                _ => unimplemented!("No text writer available for requested TextKind {:?}", kind),
-            }?;
+            let write_config = WriteConfig::<TextEncoding_1_0>::new(kind);
+            let mut writer = ApplicationWriter::with_config(write_config, buffer)?;
             writer.write_elements(elements)?;
-            writer.flush()?;
+            buffer = writer.close()?;
+            println!(
+                "Serialized as {kind:?}:\n{}",
+                std::str::from_utf8(buffer.as_slice()).unwrap()
+            );
         }
         Format::Binary => {
-            let mut binary_writer = BinaryWriterBuilder::new().build(&mut buffer)?;
+            let mut binary_writer = ApplicationWriter::<BinaryEncoding_1_0, _>::new(buffer)?;
             binary_writer.write_elements(elements)?;
-            binary_writer.flush()?;
+            buffer = binary_writer.close()?;
         }
```

Comment:
🗺️ The application writer is now used as the serializer in the Ion tests, ensuring its compliance for all (1.0) formats.